src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
No. Suppose `begin` is 0, then `end` must be strictly greater than 0. If `begin = end`, then it produces an empty (and illegal) array. Same with negative: `begin=-1` should be the same as `end =-1 = 3`, and its not allowed to have `begin=end=-1`. You would need to write `begin=-2` and `end=-1` to only include the last element in the slice.